src/snowflake/snowpark/session.py

```diff
@@ -3456,6 +3459,78 @@ def flatten(
         set_api_call_source(df, "Session.flatten")
         return df
 
+    def register_profiler(self, profiler: Profiler):
+        """Register a profiler to current session, all action are actually executed during this function"""
+        self.profiler = profiler
+        self.profiler.session = self
```

Comment:
if we pass a profiler which is created a different session, is the overwriting here on purpose?
or another way to ask the question is should one profiler only be linked with one session?

Reply:
That's a good question, I think one profiler link to only one session is a good idea since the new session could have different setting.